src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -0,0 +1,277 @@
+            }
+
+            return messageHandlerOptions;
+        }
+
+        private async Task<MessageReceiver> CreateMessageReceiverAsync(AzureServiceBusMessagePumpSettings messagePumpSettings)
+        {
+            var rawConnectionString = await messagePumpSettings.GetConnectionStringAsync();
+            var serviceBusConnectionStringBuilder = new ServiceBusConnectionStringBuilder(rawConnectionString);
+
+            MessageReceiver messageReceiver;
+            if (string.IsNullOrWhiteSpace(serviceBusConnectionStringBuilder.EntityPath))
+            {
+                // Connection string doesn't include the entity so we're using the message pump settings
+                if (string.IsNullOrWhiteSpace(messagePumpSettings.EntityName))
+                {
+                    throw new ArgumentNullException(nameof(messagePumpSettings.EntityName), "No entity name was specified while the connection string is scoped to the namespace");
+                }
+
+                messageReceiver = CreateReceiver(serviceBusConnectionStringBuilder, messagePumpSettings.EntityName, messagePumpSettings.SubscriptionName);
+            }
+            else
+            {
+                // Connection string includes the entity so we're using that instead of the message pump settings
+                messageReceiver = CreateReceiver(serviceBusConnectionStringBuilder, serviceBusConnectionStringBuilder.EntityPath, messagePumpSettings.SubscriptionName);
+            }
+
+            Namespace = messageReceiver.ServiceBusConnection?.Endpoint?.Host;
+
+            ConfigurePlugins();
+
+            return messageReceiver;
+        }
+
+        private MessageReceiver CreateReceiver(ServiceBusConnectionStringBuilder serviceBusConnectionStringBuilder, string entityName, string subscriptionName)
+        {
+            EntityPath = entityName;
+
+            if (string.IsNullOrWhiteSpace(subscriptionName) == false)
+            {
+                EntityPath = $"{EntityPath}/subscriptions/{subscriptionName}";
+            }
+
+            var connectionString = serviceBusConnectionStringBuilder.GetNamespaceConnectionString();
+            return new MessageReceiver(connectionString, EntityPath);
+        }
+
+        private void ConfigurePlugins()
+        {
+            var registeredPlugins = DefineServiceBusPlugins();
+            if (registeredPlugins != null)
+            {
+                foreach (var plugin in registeredPlugins)
+                {
+                    _messageReceiver.RegisterPlugin(plugin);
+                }
+            }
+        }
+
+        /// <summary>
+        ///     Provides capability to define Service Bus plugins to register
+        /// </summary>
+        /// <remarks>All Service Bus plugins will be registered in the same order</remarks>
+        /// <returns>List of Service Bus plugins that will be used by the message pump</returns>
+        protected virtual List<ServiceBusPlugin> DefineServiceBusPlugins()
```

Comment:
`IEnumerable` instead? Not sure if alteration of this list is required afterwards, but I guess not?

Reply:
Done